homeassistant/components/reolink/select.py

```diff
@@ -0,0 +1,120 @@
+"""This component provides support for Reolink select entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import DayNightEnum, Host, SpotlightModeEnum
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSelectEntityDescriptionMixin:
+    """Mixin values for Reolink select entities."""
+
+    method: Callable[[Host, int, str], Any]
+
+
+@dataclass
+class ReolinkSelectEntityDescription(
+    SelectEntityDescription, ReolinkSelectEntityDescriptionMixin
+):
+    """A class that describes select entities."""
+
+    supported: Callable[[Host, int], bool] = lambda api, ch: True
+    value: Callable[[Host, int], str] | None = None
+    get_options: Callable[[Host, int], Any] | None = None
+
+
+SELECT_ENTITIES = (
+    ReolinkSelectEntityDescription(
+        key="floodlight_mode",
+        name="Floodlight mode",
+        icon="mdi:spotlight-beam",
+        translation_key="floodlight_mode",
+        options=[mode.name for mode in SpotlightModeEnum],
```

Comment:
Since you're providing translations for this I think it would be a good idea that you have a translation dictionary so the key values doesn't change if the reolink API changes down the line

Reply:
The dictionary is already used, it is the SpotlightModeEnum defined in the reolink-aio library which makes the translation from API key to key used by HomeAssistant. See here: https://github.com/starkillerOG/reolink_aio/blob/10c4bbd868d7e1f8cce4167d5098214100e01a26/reolink_aio/enums.py#L6-L11

I will not change the Enum names in the reolink-aio library because that is the whole point of that Enum.